lib/schemaUtils.js

```diff
@@ -634,7 +634,7 @@ module.exports = {
     Object.assign(specComponentsAndUtils, concreteUtils.getRequiredData(spec));
 
     for (path in paths) {
-      if (paths.hasOwnProperty(path)) {
+      if (paths.hasOwnProperty(path) && (paths[path] !== undefined || paths[path] !== null)) {
```

Comment:
Better to check if `paths[path]` is object or not as that would handle more edge cases,